### PR TITLE
feat: improve error handling and state management in identifyUser and resetUser

### PR DESCRIPTION
**Pull request by jonathannorris**
## 🚀 Improved Error Handling and State Management

This PR enhances the error handling and state management capabilities of the `identifyUser` and `resetUser` methods in the DevCycle Android SDK.

#### Core Improvements
- **Enhanced `identifyUser` method**:
  - Added proper user validation with try-catch blocks before modifying client state
  - Implemented state restoration when config fetching fails
  - Added fallback to cached config when network requests fail
  - Proper error propagation while maintaining client consistency

- **Enhanced `resetUser` method**:
  - Added anonymous user ID backup and restoration on error
  - Improved error handling with proper state rollback
  - Centralized config fetching logic

#### Refactoring
- **Extracted `fetchConfigForUser` method** for better code organization and reusability
- **Added `useCachedConfigForUserWithResult` method** with explicit boolean return for better error handling
- **Updated `PopulatedUser.buildAnonymous`** to properly integrate with `DVCSharedPrefs`

#### Testing
- **Fixed SharedPreferences mocking** with proper `doNothing()` calls for void methods
- **Added comprehensive test coverage** for error scenarios, invalid users, and edge cases
- **Improved test reliability** with better async handling and timeout management

### 🎯 Benefits

1. **Better Error Recovery**: Failed operations now properly restore previous state
2. **Improved Reliability**: Network failures gracefully fall back to cached configs
3. **Enhanced User Experience**: Invalid user inputs are handled without breaking client state
4. **Robust Testing**: Comprehensive test coverage ensures reliability across scenarios

### 🧪 Test Coverage

- Invalid user validation and error handling
- Network failure scenarios with cached config fallback
- State restoration on error conditions
- Concurrent request queuing and execution
- Edge cases for anonymous user ID management

All existing tests continue to pass, ensuring backward compatibility.

### CODE REVIEW

File: android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt [hunk 1]
```diff
@@ -651,19 +656,53 @@ class DevCycleClientTests {
...
+                } else if (request.path?.contains("/v1/mobileSDKConfig") == true) {
+                    return MockResponse().setResponseCode(200).setBody(objectMapper.writeValueAsString(config))
+                }
+                return MockResponse().setResponseCode(404)
             }
-            override fun onError(t: Throwable) {}
         }
-        client.resetUser(callback)
+
+        val client = createClient("pretend-its-a-real-sdk-key", mockWebServer.url("/").toString(), user=DevCycleUser.builder().withIsAnonymous(true).build())
```
Comment (JamieSinn) [inline android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt:672]: We should be consistent and use our test token format.
dvc_mobile_token_hash

Comment (JamieSinn) [reply android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt:672]: Also the sdk should error and fail validation on this token since it's not a valid format.

File: android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt [hunk 2]
```diff
@@ -2063,6 +2102,275 @@ class DevCycleClientTests {
         latch.await(timeout, units)
         Assertions.assertEquals(0, latch.count)
     }
+
+    @Test
+    fun `identifyUser with invalid user returns error without changing client state`() {
+        val config = generateConfig("activate-flag", "Flag activated!", Variable.TypeEnum.STRING)
+        mockWebServer.enqueue(MockResponse().setResponseCode(200).setBody(objectMapper.writeValueAsString(config)))
+        
+        val client = createClient("pretend-its-a-real-sdk-key", mockWebServer.url("/").toString())
```
Comment (JamieSinn) [inline android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt:2111]: Same here

File: android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt [hunk 3]
```diff
@@ -2063,6 +2102,275 @@ class DevCycleClientTests {
...
+    fun `fetchConfigForUser queues requests when already executing`() {
+        val config = generateConfig("activate-flag", "Flag activated!", Variable.TypeEnum.STRING)
+        
+        // First request for initialization
+        mockWebServer.enqueue(MockResponse().setResponseCode(200).setBody(objectMapper.writeValueAsString(config)))
+        // Second and third requests for identifyUser calls
+        mockWebServer.enqueue(MockResponse().setResponseCode(200).setBody(objectMapper.writeValueAsString(config)))
+        mockWebServer.enqueue(MockResponse().setResponseCode(200).setBody(objectMapper.writeValueAsString(config)))
+        
+        val client = createClient("pretend-its-a-real-sdk-key", mockWebServer.url("/").toString())
```
Comment (JamieSinn) [inline android-client-sdk/src/test/java/com/devcycle/sdk/android/api/DevCycleClientTests.kt:2308]: This should probably be pulled out to a variable/constant for tests.

**Review by jsalaber — APPROVED** [review]
